woocommerce/woocommerce-gutenberg-products-block
# Add fixtures data for all entities.

This PR depends (on builds upon https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2544) I set it against master so that travis can run.

This was a huge blocker for me, glad I get it done (almost).

ِAdds fixture data for settings, taxes, coupons, shipping, payment, reviews, and a single product (needed for reviews).

those items are loaded from WC REST API, so this is done very quickly.
the code is run only once before all tests start, after finishing tests, all created items will be deleted, for reference, only the created ones will get deleted.
If you happen to create something in a test, make sure to delete it yourself or append it to the delete queue.

```js
beforeAll( async () => {
// you created a product here
// single item
global.wc.products.push( productId );
// many items
global.wc.products = [ ...productIds, global.wc.products ];
});
```
Same applies for taxes, coupons, and shipping zones, each one is exposes in an array in:

```js
global.wc {
    coupons,
    products,
    taxes,
    shippingMethods,
}
```

### How to test the changes in this Pull Request:
* Start the docker instance `npm run test:e2e:up:build`
* build the blocks to the latest version `npm run build`
* run some e2e tests `npm run test:e2e`
* When tests end, visit http://localhost:8084
* Make sure there are no coupons, products,  shipping zones, or taxes.
* Make sure fixtures actually loaded, this is done by checking the store address, it will be `60 29th Street #343`.
* Stop the docker, by running `npm run test:e2e:down`

What will follow is more tests and more fixtures for products, for now, we will still use the standard routine we had from before.
